credsweeper/app.py

```diff
@@ -299,6 +300,16 @@ def data_scan(self, data_provider: DataContentProvider, depth: int, recursive_li
             except Exception as zip_exc:
                 # too many exception types might be produced with broken zip
                 logger.error(f"{data_provider.file_path}:{zip_exc}")
+
+        elif Util.is_gzip(data_provider.data):
+            try:
+                with gzip.open(io.BytesIO(data_provider.data)) as f:
```

Comment:
Why zipped files are checked one-by-one but gzip is parsed as a whole?

Reply:
gzip contains single file only. tar is applied when necessary pack many files in gzip